test/tz_stub.py

```diff
@@ -36,5 +37,5 @@ def stub_timezone_ctx(tzname):
     os.environ["TZ"] = tzname
     time.tzset()
     yield
-    os.environ["TZ"] = old_tzname
+    os.environ["TZ"] = old_tzname  # type: ignore[assignment]
```

Comment:
What is the type of `old_tzname` here?

Reply:
here is the error message from mypy:
`Incompatible types in assignment (expression has type "Optional[str]", target has type "str")  [assignment]`

The `old_tzname` comes from `old_tzname = datetime.datetime.now(local_tz).tzname()`.
`tzname()` return type is `Optional[str]`

Reply:
Ha got it. I just checked, if we assign `None` to `os.environ` it raises a `TypeError`. So I think what we should do is `assert old_tzname is not None` right after obtaining it, to clarify our assumption and avoid the type-ignore.

Reply:
Good suggestion. Thanks!
I've updated the patch based on your suggestion.